drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -712,6 +813,50 @@ static int nrf5_configure(const struct device *dev,
 
 	case IEEE802154_CONFIG_EVENT_HANDLER:
 		nrf5_data.event_handler = config->event_handler;
+		break;
+
+#if defined(CONFIG_NRF_802154_ENCRYPTION)
+	case IEEE802154_CONFIG_MAC_KEYS:
+		nrf5_config_mac_keys(config->mac_keys);
+		break;
+
+	case IEEE802154_CONFIG_FRAME_COUNTER:
+		nrf_802154_security_global_frame_counter_set(config->frame_counter);
+		break;
+
+	case IEEE802154_CONFIG_ENH_ACK_HEADER_IE: {
```

Comment:
This `case` should not be placed inside `#if defined(CONFIG_NRF_802154_ENCRYPTION)`